src/Tests/Microsoft.NET.Sdk.Razor.Tests/BuildIntegrationTest.cs

```diff
@@ -0,0 +1,856 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyModel;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Sdk.Razor.Tests
+{
+    public class BuildIntegrationTest : SdkTest
+    {
+        public BuildIntegrationTest(ITestOutputHelper log) : base(log) {}
+
+        [Fact]
+        public void Build_SimpleMvc_UsingDotnetMSBuildAndWithoutBuildServer_CanBuildSuccessfully()
+            => Build_SimpleMvc_WithoutBuildServer_CanBuildSuccessfully();
+
+        [FullMSBuildOnlyFactAttribute]
+        public void Build_SimpleMvc_UsingDesktopMSBuildAndWithoutBuildServer_CanBuildSuccessfully()
+            => Build_SimpleMvc_WithoutBuildServer_CanBuildSuccessfully();
+
+        // This test is identical to the ones in BuildServerIntegrationTest except this one explicitly disables the Razor build server.
+        private void Build_SimpleMvc_WithoutBuildServer_CanBuildSuccessfully()
+        {
+            var testAsset = "SimpleMvc";
+            var projectDirectory = _testAssetsManager
+                .CopyTestAsset(testAsset)
+                .WithSource();
+
+            var build = new BuildCommand(projectDirectory);
+            var outputPath = build.GetOutputDirectory("net5.0", "Debug").ToString();
```

Comment:
This is a bit problematic. We usually always test against bleeding edge and use legacy tests to verify older projects continue to build. I don't want us to hardcode an exact tfm here. Do you have ideas on how to solve this?

Reply:
I added a new `RazorSdkTest` subclass that these tests build upon. It allows us to override the TFM in the projects when it is copied over. With this, we'll only need to change the TFM there to get things working.

Also updated the TFM to net6.0 for these projects.